components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java

```diff
@@ -426,12 +433,49 @@ private PostAuthnHandlerFlowStatus handleRequestFlow(HttpServletRequest request,
                     }
                     callDefaultProvisioningHandler(username, context, externalIdPConfig, localClaimValues,
                             stepConfig);
+
+                    // Trigger event to store last login time after successful authentication.
+                    if (StringUtils.isEmpty(associatedLocalUser)) {
+                        triggerEvent(username, context, username);
+                    } else {
+                        triggerEvent(associatedLocalUser, context, username);
+                    }
                 }
             }
         }
         return SUCCESS_COMPLETED;
     }
 
+    private void triggerEvent(String username, AuthenticationContext context, String emailUsername)
+            throws PostAuthenticationFailedException {
+
+        if (StringUtils.isEmpty(username)) {
+            return;
+        }
+
+        try {
+            UserRealm realm = getUserRealm(context.getTenantDomain());
+            UserStoreManager userStoreManager = getUserStoreManager(context.getExternalIdP()
+                    .getProvisioningUserStoreId(), realm, emailUsername);
+            IdentityEventService eventService = FrameworkServiceDataHolder.getInstance().getIdentityEventService();
+            Map<String, Object> eventProperties = new HashMap<>();
+            eventProperties.put(IdentityEventConstants.EventProperty.USER_NAME, username);
+            eventProperties.put(IdentityEventConstants.EventProperty.USER_STORE_MANAGER, userStoreManager);
+            eventProperties.put(IdentityEventConstants.EventProperty.OPERATION_STATUS, true);
+            Event event = new Event(IdentityEventConstants.Event.JIT_USER_SET_CLAIMS, eventProperties);
```

Comment:
Change the event name into POST_USER_JIT_PROVISION

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/4612/commits/6643361a2944594606c0a5a677c9580fbfd3eeef